### PR TITLE
fix warning and improve encrypt

### LINKED ISSUE #56: Openssl 1.1.1 salt warnings / Git for Windows interoperability
In [Git for Windows](https://git-scm.com/download/win), currently in version 2.21.0 using Bash 4.4.23 and OpenSSL 1.1.1a, the salt generation in the `clean` script causes two issues:

a) The following warning is issued by OpenSSL everytime the `openssl enc` part of the `clean` script is called: `hex string is too short, padding with zero bytes to length`
b) salt generation differs between *nix and Windows due to line-endings.

Both issues have the same solution: make the salt consist of 16 hex characters not including *any* line-ending characters. Pull request on the way.

### LINKED ISSUE #59: hex string is too short, padding with zero bytes to length *** WARNING : deprecated key derivation used. Using -iter or -pbkdf2 would be better.
Defalult encrypt is always warning,plase add -pbkdf2 -iter 1024 to update

openssl enc -aes-256-cbc -md sha512 -pbkdf2 -iter 1024 -salt -in InputFilePath -out OutputFilePath

follow by https://crypto.stackexchange.com/questions/20941/why-shouldnt-i-use-ecb-encryption

### PR DESCRIPTION
**Pull request by ricardo0624**
fix #56 
fix #59 

1. fix waring like above
2. import default password 30 to 32,as 2 power 5
3. -md5 change to sha1 to improve encrypt(sha1 is more stronger than md5)

maybe it is not incompatible with last versions

I have simply test it, it works well

### CODE REVIEW

File: transcrypt [hunk 1]
```diff
@@ -292,8 +292,8 @@ save_helper_scripts() {
 		  else
 		    cipher=$(git config --get --local transcrypt.cipher)
 		    password=$(git config --get --local transcrypt.password)
-		    salt=$(openssl dgst -hmac "${filename}:${password}" -sha256 "$filename" | tail -c 16)
-		    ENC_PASS=$password openssl enc -$cipher -md MD5 -pass env:ENC_PASS -e -a -S "$salt" -in "$tempfile"
+		    salt=$(openssl dgst -hmac "${filename}:${password}" -sha256 "$filename" | tail -c 17)
```
Comment (elasticdog) [inline transcrypt]: I went ahead and merged in #57 to fix the issue in #56 in a cross-platform manner, so this change can be reverted if you merge in the latest from the master branch.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.idea/**
+
+transcrypt.iml
```
[the remark this answers is not in the document]
Comment (ricardo0624) [reply .gitignore]: thanks for taking your time,this project was really simple and great,I will remember to exclude the project-specific file

**Comment by ricardo0624** [comment]
backward incompatible is very import ,but '-iter' option openssl is really recommended,so I suggest that in new project , write transcrypt version to git,then if the transcrypt version is exist and correct ,then use new function,else use the old function.
security is very import ,so why users use this project, but also hack technology is developing,so I think it is necessarily to improve the encrypt strength to project their projects as they are care about that and why use this project to encrypt.

**Comment by elasticdog** [comment]
Definitely agree that we should modernize the algorithm to increase security and figure out the proper migration path, whether that means adding version checking into the script or something else...that said, it's not a decision I want to rush.

**Comment by ricardo0624** [comment]
> Definitely agree that we should modernize the algorithm to increase security and figure out the proper migration path, whether that means adding version checking into the script or something else...that said, it's not a decision I want to rush.

yeah,couldn't agree more,no need to rush

**Pull request closed without merging**